posthog/hogql/functions/mapping.py

```diff
@@ -2045,3 +2098,46 @@ def find_hogql_posthog_function(name: str) -> Optional[HogQLFunctionMeta]:
 def is_allowed_parametric_function(name: str) -> bool:
     # No case-insensitivity for parametric functions
     return name in HOGQL_PERMITTED_PARAMETRIC_FUNCTIONS
+
+
+def get_correct_function_name(name: str) -> str:
+    if HOGQL_CLICKHOUSE_FUNCTIONS.get(name) or HOGQL_AGGREGATIONS.get(name) or HOGQL_POSTHOG_FUNCTIONS.get(name):
+        return name
+
+    name_lower = name.lower()
+    for key in HOGQL_CLICKHOUSE_FUNCTIONS:
+        if key.lower() == name_lower:
+            return key
+
+    for key in HOGQL_AGGREGATIONS:
+        if key.lower() == name_lower:
+            return key
+
+    for key in HOGQL_POSTHOG_FUNCTIONS:
+        if key.lower() == name_lower:
+            return key
+
+    return name
+
+
+def normalize_hogql_function_names(query: str) -> str:
+    """Normalize function names in a HogQL query to use correct casing.
+
+    This function uses regex to find function calls and normalizes their casing
+    based on the known HogQL function mappings.
+    """
+    # Pattern to match function calls: word followed by opening parenthesis
+    # This captures the function name before the opening parenthesis
+    function_pattern = r"\b([a-zA-Z_][a-zA-Z0-9_]*)\s*\("
+
+    def replace_function_name(match):
+        function_name = match.group(1)
+
+        if function_name.upper() in SQL_KEYWORDS:
+            return match.group(0)
```

Comment:
How are keywords coming into play here - I thought this was just replacing function names?

Reply:
Ahh I see - more reason to not use regex then!